### PR TITLE
users-api-spec

### PR DESCRIPTION
**Pull request by AmberYLopez**

### CODE REVIEW

File: spec/users-api-spec.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+'use strict';
+import request from 'supertest';
+import app from '../app/server';
+import finish from './finish';
+
+fdescribe('users-api', () => {
```
Comment (freewind) [inline spec/users-api-spec.js]: fdescribe -> describe

File: spec/users-api-spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  it('注册成功', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: 'O1o2y354', password: '123456'})
+      .expect({httpCode: 201, message: '注册成功'}, function (err, Users) {
```
Comment (freewind) [inline spec/users-api-spec.js]: `.expect({httpCode: 201, message: '注册成功'}, finish(done)`

Comment (freewind) [inline spec/users-api-spec.js]: Users -> users

File: spec/users-api-spec.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        finish(done)(err);
+      });
+
+  });
+  it('用户名位数错误(5位)', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '12345', password: '123456'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        finish(done)(err);
+      })
+  });
+
+  it('用户名位数错误(21位)', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '123456789123456789456', password: '123456'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        finish(done)(err);
+      })
+  });
+  it('密码位数错误(5位)', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '123456', password: '12345'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
+        finish(done)(err);
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      })
+  });
+  it('密码位数错误(7位)', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '123456', password: '1234567'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        finish(done)(err);
+      })
+  });
+  it('用户名密码位数都错误', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '1234', password: '1234567'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
+        finish(done)(err);
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      })
+  });
+  it('用户名格式错误', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '1234*689fsf', password: '123456'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
+        finish(done)(err);
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      })
+  });
+  it('密码格式都错误', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: '1234jvfdh', password: '1234_5'})
+      .expect({httpCode: 400, message: '用户名只能是6-20位数字、字母组成，密码只能是6位数字'}, function (err, Users) {
+        finish(done)(err);
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+      })
+  });
+  it('用户名已存在', (done) => {
+    request(app)
+      .post('/api/users')
+      .send({userName: 'O1o2y354', password: '123456'})
+      .expect({httpCode: 409, message: '该用户已存在'}, function (err, data) {
+        finish(done)(err);
```
Comment (freewind) [inline spec/users-api-spec.js]: 同前

File: spec/users-api-spec.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
+'use strict';
+import request from 'supertest';
+import app from '../app/server';
+import finish from './finish';
+
+fdescribe('users-api', () => {
+
```
Comment (freewind) [inline spec/users-api-spec.js]: beforeEach()

**Comment by Yvette14** [comment]
:+1:

**Pull request closed without merging**